core/iwasm/libraries/lib-wasi-threads/tid_allocator.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include "tid_allocator.h"
+#include "wasm_export.h"
+#include "bh_log.h"
+
+#define MIN(a, b) (((a) < (b)) ? (a) : (b))
+
+struct tidAllocator {
+    int32 *ids;  // Array used to store the stack
+    uint32 size; // Stack capacity
+    uint32 pos;  // Index of the element after the stack top
+};
+
+TidAllocator *
+tid_allocator_init()
+{
+    TidAllocator *tid_allocator = wasm_runtime_malloc(sizeof(TidAllocator));
+    tid_allocator->size = MIN(TID_ALLOCATOR_INIT_SIZE, TID_MAX - TID_MIN + 1);
+    tid_allocator->pos = tid_allocator->size;
+    tid_allocator->ids =
+        wasm_runtime_malloc(tid_allocator->size * sizeof(int32));
+    if (tid_allocator->ids == NULL)
+        return NULL;
+
+    for (int64 i = tid_allocator->pos - 1; i >= 0; i--)
```

Comment:
int32 is enough

Reply:
but `tid_allocator->pos` is `uint32`, it may not fit into `int32`